composer/loggers/progress_bar_logger.py

```diff
@@ -56,10 +65,21 @@ def __init__(
             unit=unit,
         )
 
-    def log_data(self, data: Dict[str, Any]):
-        formatted_data = {k: format_log_data_value(v) for (k, v) in data.items() if k in self.keys_to_log}
-        self.metrics.update(formatted_data)
-        self.pbar.set_postfix(self.metrics)
+    def log_data(self, state: State):
+        dataloader_label = state.dataloader_label
+        if self.metric_string is None:
+            formatted_data = {}
+            curr_metrics = state.current_metrics.get(dataloader_label, {})
+            for current_metric, v in curr_metrics.items():
+                formatted_data[current_metric] = format_log_data_value(v)
+            if self.is_train and state.timestamp.batch_in_epoch != 0:
+                formatted_data['loss/train'] = format_log_data_value(state.loss)
+            self.pbar.set_postfix(formatted_data)
+        else:
+            current_metrics = state.current_metrics.get(dataloader_label, {})
+            metric_string = self.metric_string.format_map(FormatDict(current_metrics))
+            metric_string = metric_string.format_map(FormatDict(state.__dict__))
```

Comment:
Does `state.__dict__` work with state elements defined as `@property`?

Reply:
Yes and no. For example it won't work for `dataloader_label`, but it will work for `_dataloader_label`. I can mention this in the doc strings.